### PR TITLE
test: validate dynamically resized volumes

### PR DESCRIPTION
**Pull request by kassarl**
**What type of PR is this?**

/kind test
**What this PR does / why we need it**:
adds a test to validate dynamic resizing on zones that support LiveResize
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

File: test/e2e/testsuites/dynamically_provisioned_resize_volume_tester.go [hunk 1]
```diff
@@ -47,6 +48,7 @@ type DynamicallyProvisionedResizeVolumeTest struct {
 	StorageClassParameters map[string]string
 	Pod                    PodDetails
 	Volume                 VolumeDetails
+	DetachVM               bool
```
Comment (edreed) [inline test/e2e/testsuites/dynamically_provisioned_resize_volume_tester.go:51]: Consider `ResizeOffline` to better reflect the scenario. The VM isn't what's being detached here.

**Comment by andyzhangx** [comment]
```
# sigs.k8s.io/azuredisk-csi-driver/test/e2e
vet: test/e2e/dynamic_provisioning_test.go:824:4: unknown field DetachVM in struct literal
make: *** [Makefile:79: verify] Error 2
```

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**